Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -146,9 +146,12 @@ extension BrowserViewController: WKNavigationDelegate {
             // Weird behavior here with `targetFram` and `sourceFrame`, on refreshing page `sourceFrame` is not nil (it is non-optional)
             //  however, it is still an uninitialized object, making it an unreliable source to compare `isMainFrame` against.
             //  Rather than using `sourceFrame.isMainFrame` or even comparing `sourceFrame == targetFrame`, a simple URL check is used.
+            // No adblocking logic is be used on session restore urls. It uses javascript to retrieve the
+            // request then the page is reloaded with a proper url and adblocking rules are applied.
             if
                 let mainDocumentURL = navigationAction.request.mainDocumentURL,
                 mainDocumentURL == url,
+                !url.isSessionRestoreURL,
```

Comment:
Not 100% sure about placement of this check. Alternatively I could place it only before applying script blocking, but I do not think there is any case where we want to do ad block operations on session restore htmls

Reply:
Yeah, I think this is correct.